src/modules/dlang.rs

```diff
@@ -0,0 +1,143 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::dlang::DLangConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the current D version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("dlang");
+    let config = DLangConfig::try_load(module.config);
+    let is_dlang_project = context
+        .try_begin_scan()?
+        .set_extensions(&config.detect_extensions)
+        .set_files(&config.detect_files)
+        .is_match();
+
+    if !is_dlang_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => context
+                    .exec_cmd("dmd", &["--version"])
+                    .and_then(|output| parse_dlang_version(output.stdout.trim()))
+                    .map(Ok),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `dlang`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn parse_dlang_version(dlang_version: &str) -> Option<String> {
+    let version = dlang_version
+        // split into lines
+        .lines()
+        // first line
+        .next()?
+        // split by whitespace
+        .split_whitespace()
+        // get 3rd index once splitted
+        .nth(3)?;
+
+    Some(version.to_owned())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::parse_dlang_version;
+    use crate::test::ModuleRenderer;
+    use ansi_term::Color;
+    use std::fs::File;
+    use std::io;
+
+    #[test]
+    fn test_parse_dlang_version() {
+        const OUTPUT: &str = "DMD32 D Compiler v2.096.0-dirty
+        Copyright (C) 1999-2021 by The D Language Foundation, All Rights Reserved written by Walter Bright\n";
```

Comment:
Is this supposed to be indented?

Reply:
I'm confused, that's what dmd --version returns.

Reply:
I'm asking about the space between dirty and copyright.
```diff
+        const OUTPUT: &str = "DMD32 D Compiler v2.096.0-dirty
+        Copyright (C) 1999-2021 by The D Language Foundation, All Rights Reserved written by Walter Bright\n";
```

Reply:
Oh it's a new line